src/MAT_HDF5.jl

```diff
@@ -28,15 +28,15 @@
 
 module MAT_HDF5
 
-using HDF5
+using HDF5, SparseArrays
 
 import Base: read, write, close
 import HDF5: names, exists, HDF5ReferenceObj, HDF5BitsKind
 
 const HDF5Parent = Union{HDF5File, HDF5Group}
 const HDF5BitsOrBool = Union{HDF5BitsKind,Bool}
 
-type MatlabHDF5File <: HDF5.DataFile
+mutable struct MatlabHDF5File <: HDF5.DataFile
```

Comment:
I think this can simple be a struct

Reply:
Ok. Do you know if the other one or two occurrences of mutable structs can be constant as well?

Reply:
I believe they should all be struct's

Reply:
Nope. Turns out both the `MatlabHDF5File` and `Matlabv5File` need to be mutable.